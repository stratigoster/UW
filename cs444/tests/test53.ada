PACKAGE foo IS
  type Ord is (LT, GT, EQ, LEQ, GEQ);
  type List is array (1..10,2..20) of string;
  l : ord := Ord'Last'Pred;
  f : ord := Ord'First;
  aList : List;

  i : integer := List'Length;
  a : integer := i'pos;

  c : boolean := true;
  b : integer := c'pos'pred;

  d : integer := f'pos;

  pi : float := 3.141592653589793238;
  p : integer := pi'ceiling;
  q : integer := pi'floor;
  procedure foo (x,y,z:integer);
  procedure foo;
  procedure foo (a,b,c:string; d,e,f:ord);
begin
  declare
    procedure foo(a,b,z:integer) is
    begin
      c := a = b and then b = z;
    end foo;
    i : integer := foo(1,2,3);
--                 ^^^^^^^^^^
-- procedures don't have return values
  begin
    for x in 1..5 loop 
      i := i + 1;
    end loop;
    null;
  end;
END foo;
