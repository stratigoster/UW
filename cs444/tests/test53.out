***ERROR***:28:7:	Incompatible types in assignment of ` void ' to ` universal integer '

------
Compilation error
------
